src/libraries/System.Net.Http/src/System/Net/Http/MultipartContent.cs

```diff
@@ -267,16 +274,18 @@ private async Task<Stream> CreateContentReadStreamAsyncCore(CancellationToken ca
                     HttpContent nestedContent = _nestedContent[contentIndex];
                     streams[streamIndex++] = EncodeStringToNewStream(SerializeHeadersToString(scratch, contentIndex, nestedContent));
 
-                    Stream readStream = nestedContent.TryReadAsStream() ?? await nestedContent.ReadAsStreamAsync(cancellationToken).ConfigureAwait(false) ?? new MemoryStream();
+                    Stream readStream = async ?
+                        nestedContent.TryReadAsStream() ?? await nestedContent.ReadAsStreamAsync(cancellationToken).ConfigureAwait(false) ?? new MemoryStream() :
+                        nestedContent.ReadAsStream();
```

Comment:
All of these ? and ?? operators makes it hard to understand order of precedence here.  Can you please add some parens to help make it clear what's going on?

Reply:
Split up into `if/else`, defaulting to empty `MemoryStream` moved as the last step.